File: Programmers/HEROHJK/12943.swift
import Foundation

func solution(_ num:Int) -> Int {
    var num = num
    var count = 0
    while num > 1 && count < 500 {
        if num % 2 == 0 {
            num /= 2
        } else {
            num = num * 3 + 1
        }
        
        count += 1
    }
    
    return (count == 500 && num != 1) ? -1 : count
}

File: Programmers/NOMORE12/programmers-lv-1/level_1/12943.js
function solution(num) {
  var answer = 0;
  let number = num;
  for (let i = 0; i < 500; i++) {
    if (i === 499) answer = -1;
    if (number === 1) {
      answer = i;
      break;
    }
    number = number % 2 === 0 ? number / 2 : number * 3 + 1;
  }
  return answer;
}

console.log(solution(16))